### ISSUE TITLE
False positive for undefined variable in preg_replace

### ISSUE THREAD

**Issue by kkmuffme** [sirbrillig/phpcs-variable-analysis#60]
Simple sample:

```php
$nav_menu = 'hello world';
$selector = 'good morning';
$nav_menu = preg_replace( '/(hello \w+)/', "$selector$1", $nav_menu, 1 );
```

It will report the $1 as undefined variable. within preg_replace variables that match \$\d+ should only be reported as undefined, if there is no matching group in the search of the regex.

**Comment by sirbrillig** [comment]
That's a very good observation, thanks! For later reference, the documentation for the second argument of `preg_replace()` reads:

> **replacement** may contain references of the form \\n or $n, with the latter form being the preferred one.

This could be handled as a special case: ignore undefined variables between the parenthesis of `preg_replace()` if the names of those variables are entirely numeric. 

[On the other hand](http://php.net/manual/en/language.variables.basics.php), 

> A valid variable name starts with a letter or underscore, followed by any number of letters, numbers, or underscores

So actual PHP variables cannot ever start with a number, so perhaps we could solve this simply by always ignoring variables that start with a number. If they exist and are illegal, some other sniff should detect them first. I think I like this solution better because it's simpler to implement and actually should decrease the amount of processing this sniff has to do.
